### PR TITLE
docs(ad-api): add use cases

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add the use cases of AD API.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The Reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: docs/design/autoware-interfaces/ad-api/use-cases/bus-service.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Use Case of Bus Service
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/ad-api/use-cases/bus-service.md:1]: [style] We use sentence case.

```suggestion
# Use case of bus service
```

File: docs/design/autoware-interfaces/ad-api/use-cases/taxi-service.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Use Case of Bus Service
```
Comment (kenji-miyake) [inline docs/design/autoware-interfaces/ad-api/use-cases/taxi-service.md:1]: Same here.

```suggestion
# Use cases of bus service
```

**Review by kenji-miyake — COMMENTED** [review]
Add comments about the styles.

**Comment by kenji-miyake** [comment]
if you fix styles, I think it's okay to merge this as a first draft.

**Review by kenji-miyake — DISMISSED** [review]

**Comment by mitsudome-r** [comment]
pre-commit.ci run

**Review by mitsudome-r — APPROVED** [review]
LGTM

**Pull request merged**